packages/resource-deployment/scripts/setup-all-pools-for-batch.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+set -eo pipefail
+
+
+export resourceGroupName
+export batchAccountName
+export pool
+export keyVault
+export resourceName
+export systemAssignedIdentities
+export principalId
+export enableSoftDeleteOnKeyVault
+
+exitWithUsageInfo() {
+    echo "
+Usage: $0 -r <resource group>
+"
+    exit 1
+}
+
+function waitForProcesses() {
+    local processesToWaitFor=$1
+
+    list="$processesToWaitFor[@]"
+    for pid in "${!list}"; do
+        echo "Waiting for process with pid $pid"
+        wait $pid
+        echo "Process with pid $pid exited"
+    done
+}
+
+function setupPools() {
+    # Enable managed identity on Batch pools
+    pools=$(az batch pool list --query "[].id" -o tsv)
+
+    echo "Setup system identity for created pools"
+    for pool in $pools; do
```

Comment:
can this run in parallel?

Reply:
nope. key vault access policy cannot be set in parallel for the same keyvault. It throws conflict error & ends up removing some access policy.